compiler/damlc/daml-stdlib-src/DA/Map.daml

```diff
@@ -22,10 +22,22 @@ module DA.Map where
 -- This will give access to the `Map` type, and the various operations
 -- as `M.lookup`, `M.insert`, `M.fromList`, etc.
 --
--- Note that the `Map k v` functions use the built-in equality for the
--- type `k`, not the `Eq k` instance. Nevertheless, the `Eq k` constraint
--- is usually required to prevent runtime errors.
+-- `Map k v` internally uses the built-in order for the type `k`.
+-- This means that keys that contain functions are not comparable
+-- and will result in runtime errors. To prevent this, the `Ord k`
+-- instance is required for most map operations. It is recommended to
+-- only use `Map k v` for key types that have an `Ord k` instance
+-- that is derived automatically using `deriving`:
 --
+-- ```
+-- data K = ...
+--   deriving (Eq, Ord)
+-- ```
+--
+-- This includes all built-in types that aren't function types: `Int`,
+-- `Text`, `Bool`, `(a, b)` assuming `a` and `b` have default `Ord`
+-- instances, `TextMap v` assuming `v` has default `Ord` instance,
```

Comment:
Do we still want to mention TextMap given that this is going to go away soonish?

Reply:
Good point, I'll update and include more useful examples like `Optional` and `List`.